### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/libmdbx/commit/7da9f9396f6d4d604f4efb9777e484869a73a616#comments

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/libmdbx/commit/ed3ac8478728b545a50f68357008a70f260aaad3

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -224,7 +224,7 @@ the user's point of view.
   > _libmdbx_ database format depends only on the [endianness](https://en.wikipedia.org/wiki/Endianness) but not on the [bitness](https://en.wiktionary.org/wiki/bitness).
 
 6. LIFO policy for Garbage Collection recycling. This can significantly increase write performance due write-back disk cache up to several times in a best case scenario.
-  > LIFO means that for reuse will be taken the latest becames unused pages.
+  > LIFO means that for reuse will be taken the latest becomes unused pages.
```
Comment (jsoref) [inline README.md:227]: I'm uncertain about this

File: cmake/compiler.cmake [hunk 2]
```diff
@@ -231,7 +231,7 @@ else()
   check_compiler_flag("-ggdb" CC_HAS_GGDB)
   check_compiler_flag("-fvisibility=hidden" CC_HAS_VISIBILITY)
   check_compiler_flag("-march=native" CC_HAS_ARCH_NATIVE)
-  check_compiler_flag("-Og" CC_HAS_DEBUG_FRENDLY_OPTIMIZATION)
+  check_compiler_flag("-Og" CC_HAS_DEBUG_FRIENDLY_OPTIMIZATION)
```
Comment (jsoref) [inline cmake/compiler.cmake:234]: I'm uncertain about this -- I can't see anything using it

File: src/lck-windows.c [hunk 3]
```diff
@@ -195,7 +196,7 @@ MDBX_INTERNAL_FUNC int mdbx_rdt_lock(MDBX_env *env) {
 
 MDBX_INTERNAL_FUNC void mdbx_rdt_unlock(MDBX_env *env) {
   if (env->me_lfd != INVALID_HANDLE_VALUE) {
-    /* transite from S-E (locked) to S-? (used), e.g. unlock upper-part */
+    /* translate from S-E (locked) to S-? (used), e.g. unlock upper-part */
```
Comment (jsoref) [inline src/lck-windows.c:199]: I'm very uncertain about this

File: src/mdbx_dump.c [hunk 4]
```diff
@@ -224,7 +224,7 @@ static void usage(void) {
           "  -a\t\tdump main DB and all subDBs,\n"
           "    \t\tby default dump only the main DB\n"
           "  -s\t\tdump only the named subDB\n"
-          "  -r\t\trescure mode (ignore errors to dump corrupted DB)\n",
+          "  -r\t\trescue mode (ignore errors to dump corrupted DB)\n",
```
Comment (jsoref) [inline src/mdbx_dump.c:227]: This user visible

File: test/osal-windows.cc [hunk 5]
```diff
@@ -186,10 +186,10 @@ bool actor_config::osal_deserialize(const char *str, const char *end,
 //-----------------------------------------------------------------------------
 
 typedef std::pair<HANDLE, actor_status> child;
-static std::unordered_map<mdbx_pid_t, child> childs;
+static std::unordered_map<mdbx_pid_t, child> children;
```
Comment (jsoref) [inline test/osal-windows.cc:189]: I'm not uncertain about this

Comment (erthink) [reply test/osal-windows.cc:189]: @jsoref, thank you very much.
I will definitely accept this PR.

There are indeed inaccuracies in the corrections, so I will have to carefully review the proposed changes.
Some of the inaccuracies are due to the use of jargon, bad names, and abbreviations.
Nonetheless most of these cases are now clearly visible and there is a reason to correct everything.

In the near future, I will try to apply your changes, but in the devel branch, so as not to cause unnecessary difficulties during the subsequent merge.

Thanks again.
Regards.

**Comment by erthink** [comment]
Related to https://github.com/erthink/libmdbx/issues/87.

**Pull request closed without merging**